mmocr/datasets/pipelines/loading.py

```diff
@@ -66,3 +67,39 @@ def _load_masks(self, results):
         results['gt_masks'] = gt_masks
         results['mask_fields'].append('gt_masks')
         return results
+
+
+@PIPELINES.register_module()
+class LoadImageFromMat(LoadImageFromFile):
+    """Load an image from cv Mat.
+
+    Similar with :obj:`LoadImageFromFile`, but the image read from
+    ``results['img']``, which is np.ndarray.
+    """
+
+    def __call__(self, results):
+        """Call functions to add image meta information.
+
+        Args:
+            results (dict): Result dict with Webcam read image in
+                ``results['img']``.
+
+        Returns:
+            dict: The dict contains loaded image and meta information.
+        """
+
+        img = results['img']
+        if self.to_float32:
+            img = img.astype(np.float32)
+        if self.color_type == 'grayscale' and img.shape[2] == 3:
+            img = mmcv.bgr2gray(img, keepdim=True)
+        if self.color_type == 'color' and img.shape[2] == 1:
+            img = mmcv.gray2bgr(img)
+
+        results['filename'] = None
```

Comment:
Why drops the 'filename' info (if exists)

Reply:
Just a cropped image of numpy array, no filename. This func is modified from `LoadImageFromWebcam` in mmdetection, which not support color conversion (maybe a bug)

Reply:
Okay fine. The name contains "Load" so this is reasonable